File: branches/Andean/Fantasy.Base/Collections/DictionaryExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy
{
    public static class DictionaryExtensions
    {
        public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dict, TKey key, TValue @default)
        {
            TValue rs;
            if (dict.TryGetValue(key, out rs))
            {
                return rs;
            }
            else
            {
                return @default;
            }
        }

        public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dict, TKey key)
        {
            TValue rs;
            if (dict.TryGetValue(key, out rs))
            {
                return rs;
            }
            else
            {
                return default(TValue);
            }
        }
   
    }
}

File: AutoConverted/Fantasy.Base/Collection/DictionaryExtensions.java
package Fantasy;

public final class DictionaryExtensions
{
//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dict, TKey key, TValue @default)
	public static <TKey, TValue> TValue GetValueOrDefault(java.util.Map<TKey, TValue> dict, TKey key, TValue _default)
	{
		TValue rs = null;
		if ((rs = dict.get(key)) != null)
		{
			return rs;
		}
		else
		{
			return _default;
		}
	}

//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dict, TKey key)
	public static <TKey, TValue> TValue GetValueOrDefault(java.util.Map<TKey, TValue> dict, TKey key)
	{
		TValue rs = null;
		if ((rs = dict.get(key)) != null)
		{
			return rs;
		}
		else
		{
			return null;
		}
	}

}